lib/rubocop/cop/layout/redundant_line_break.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Layout
+      # This cop checks whether certain expressions, e.g. method calls, that could fit
+      # completely on a single line, are broken up into multiple lines unnecessarily.
+      #
+      # @example
+      #   # bad
+      #   foo(
+      #     a,
+      #     b
+      #   )
+      #
+      #   # good
+      #   foo(a, b)
+      class RedundantLineBreak < Cop
+        include CheckAssignment
+
+        MSG = 'Redundant line break detected.'
+
+        def on_send(node) # rubocop:todo Metrics/CyclomaticComplexity
+          # Include "the whole expression".
+          node = node.parent while convertible_block?(node) ||
+                                   node.parent.is_a?(RuboCop::AST::BinaryOperatorNode) ||
+                                   node.parent&.send_type?
+
+          return unless offense?(node) && !part_of_ignored_node?(node)
+
+          add_offense(node)
+          ignore_node(node)
+        end
+
+        def check_assignment(node, _rhs)
+          return unless offense?(node)
+
+          add_offense(node)
+          ignore_node(node)
+        end
+
+        def autocorrect(node)
+          ->(corrector) { corrector.replace(node.source_range, to_single_line(node.source).strip) }
+        end
+
+        private
+
+        def offense?(node)
+          !single_line?(node) && !too_long?(node) && suitable_as_single_line?(node)
+        end
+
+        def suitable_as_single_line?(node)
+          !comment_within?(node) &&
+            node.each_descendant(:if, :case, :kwbegin, :def).none? &&
+            node.each_descendant(:dstr, :str).none?(&:heredoc?) &&
+            node.each_descendant(:begin).none? { |b| b.first_line != b.last_line }
+        end
+
+        def convertible_block?(node)
```

Comment:
I wonder if we don't need something like this for convertable `def`s.

Reply:
Possibly in the future, but currently the cop doesn't have any support for joining method definition lines, so the problem doesn't come up. Did I understand your comment correctly?